bugfix: resolve crash with FBX Settings (#4813)

pollend opened the pull request:
This works around the crash by statically linking editor common to EditorAssetImporter. There is some kind of mis-configuration where AZ::SystemAllocator is not correctly configured in the dynamic binary. I would like to investigate this more but this should address the crash atleast. 

how do static fields work between shared libraries? InitializeEditorCommonISystem is called in IEditorImpl.cpp but why is no allocator defined.  

issue: https://github.com/o3de/o3de/issues/4813

Signed-off-by: Michael Pollind <mpollind@gmail.com>

amznestebanpapp commented on Code/Editor/Plugins/EditorAssetImporter/CMakeLists.txt:
```diff
@@ -36,7 +36,8 @@ ly_add_target(
             Legacy::CryCommon
             Legacy::EditorLib
-            Legacy::EditorCommon
+            # TODO: enviroment is misconfigured causing crash so link to common statically
+            Legacy::EditorCommon.Static
```
This means that when EditorCommon is loaded, the environment is not properly attached. This is happening during the call to `InitializeEditorCommonISystem` which is called from `CEditorImpl::SetGameEngine`

I think we need to tackle the root problem and figure out why the environment is misconfigured. Besides We will get a bigger binary on this plugin and can have multiple copies of some variables (which can cause runtime issues).

pollend replied:
@amznestebanpapp just hit me looking at the stack trace.

libEditorCommon.so --> libEditorLib.so so you setup the environment for libEditorLib.so not editorCommon 
```
* thread #1, name = 'Editor', stop reason = breakpoint 5.1
  * frame #0: 0x00007ffff0840330 libEditorLib.so`::ModuleInitISystem(pSystem=0x0000000006d214f0, moduleName="EditorCommon") at platform_impl.cpp:79:9
    frame #1: 0x00007fffee089a0c libEditorCommon.so`InitializeEditorCommonISystem(pSystem=0x0000000006d214f0) at EditorCommon.cpp:31:5
    frame #2: 0x00007ffff063861f libEditorLib.so`CEditorImpl::SetGameEngine(this=0x0000000005db7710, ge=0x00000000035f8f80) at IEditorImpl.cpp:360:5
    frame #3: 0x00007ffff072d347 libEditorLib.so`CCryEditApp::InitGameSystem(this=0x0000000002e28240, hwndForInputSystem=0x0000000004000014) at CryEdit.cpp:1059:16
    frame #4: 0x00007ffff0732d4c libEditorLib.so`CCryEditApp::InitInstance(this=0x0000000002e28240) at CryEdit.cpp:1683:34
    frame #5: 0x00007ffff073b640 libEditorLib.so`::CryEditMain(argc=3, argv=0x00007fffffffd818) at CryEdit.cpp:4188:57
    frame #6: 0x000000000043d90a Editor`decltype(f=0x00007fffffffd698, args=0x00007fffffffd6d4, args=0x00007fffffffd6c8)(int, char**)>(fp)(InvokeTraits::forward<int&>(fp0), InvokeTraits::forward<char**&>(fp0))) AZStd::Internal::INVOKE<int (*&)(int, char**), int&, char**&>(int (*&)(int, char**), int&, char**&) at invoke_traits.h:208:20
    frame #7: 0x000000000043d7c7 Editor`AZStd::invoke_result<int (*&)(int, char**), int&, char**&>::type AZStd::invoke<int (f=0x00007fffffffd698, args=0x00007fffffffd6d4, args=0x00007fffffffd6c8)(int, char**), int&, char**&>(int (*&)(int, char**), int&, char**&) at invoke.h:38:16
    frame #8: 0x000000000043d58f Editor`main(argc=3, argv=0x00007fffffffd818) at main.cpp:32:15
    frame #9: 0x00007ffff684bfd0 libc.so.6`__libc_start_call_main(main=(Editor`main at main.cpp:13), argc=3, argv=0x00007fffffffd818) at libc_start_call_main.h:58:16
    frame #10: 0x00007ffff684c07d libc.so.6`__libc_start_main_impl(main=(Editor`main at main.cpp:13), argc=3, argv=0x00007fffffffd818, init=0x0000000000000000, fini=<unavailable>, rtld_fini=<unavailable>, stack_end=0x00007fffffffd808) at libc-start.c:409:3
    frame #11: 0x000000000043d3a5 Editor`_start + 37

```

amznestebanpapp replied:
> @amznestebanpapp just hit me looking at the stack trace.
> 
> libEditorCommon.so --> libEditorLib.so so you setup the environment for libEditorLib.so not editorCommon

Someone else hit the same issue (or was the same person?): https://discord.com/channels/805939474655346758/816043622558335046/905115997478981742

pollend replied:
> > @amznestebanpapp just hit me looking at the stack trace.
> > libEditorCommon.so --> libEditorLib.so so you setup the environment for libEditorLib.so not editorCommon
> 
> Someone else hit the same issue (or was the same person?): https://discord.com/channels/805939474655346758/816043622558335046/905115997478981742

that's me.

pollend replied:
that seems to have resolve the crash @amznestebanpapp I also took the chance to remove the functions that call into these symbols since they're not exported anymore.

amznestebanpapp commented on Code/Legacy/CrySystem/SystemInit.cpp:
```diff
@@ -445,18 +441,6 @@ AZStd::unique_ptr<AZ::DynamicModuleHandle> CSystem::LoadDLL(const char* dllName)
         return handle;
     }
 
-    //////////////////////////////////////////////////////////////////////////
```
nice, I will add a ticket to remove this function altogether, the only cry module left is CrySystem which is the one doing the loading.
there is another call to `LoadDLL` (`CSystem::InitVTuneProfiler` that can also be removed, but would remove all the `PROFILE_WITH_VTUNE` integration)
Created:
- https://github.com/o3de/o3de/issues/5261 to remove vtune integration in cry
- https://github.com/o3de/o3de/issues/5262 to remove all this dll loading functions from Cry

amznestebanpapp commented:
@pollend we need one more reviewer, the validations passed

pollend commented:
> @pollend we need one more reviewer, the validations passed

hmm, what could I do. I don't think I can add reviewers to this PR at most I can just ping people here or drop it in the discord.

@hultonha

amznestebanpapp commented:
> > @pollend we need one more reviewer, the validations passed
> 
> hmm, what could I do. I don't think I can add reviewers to this PR at most I can just ping people here or drop it in the discord.
> 
> @hultonha

I added some sigs to review and added the tag, it should get some eyes within 24hs.
I was just letting you know the state of the PR. We need to have 2 reviewer approvals and to pass all validations. We are just missing one reviewer to be able to merge the PR.

pollend commented:
> > > @pollend we need one more reviewer, the validations passed
> > 
> > 
> > hmm, what could I do. I don't think I can add reviewers to this PR at most I can just ping people here or drop it in the discord.
> > @hultonha
> 
> I added some sigs to review and added the tag, it should get some eyes within 24hs. I was just letting you know the state of the PR. We need to have 2 reviewer approvals and to pass all validations. We are just missing one reviewer to be able to merge the PR.

sounds great

amznestebanpapp commented:
everything ready, will merge tomorrow morning PDT

hultonha reviewed (approved):
LGTM :+1:

moraaar commented:
This was merged into development. Should this be merged into stabilization/2110 too?

amznestebanpapp commented:
> This was merged into development. Should this be merged into stabilization/2110 too?

yeah, we should, do you want to cherry-pick it?

moraaar commented:
> > This was merged into development. Should this be merged into stabilization/2110 too?
> 
> yeah, we should, do you want to cherry-pick it @moraaar?

Done. Cherry pick PR: https://github.com/o3de/o3de/pull/5365
